lib/error-handler.js

```diff
@@ -62,13 +63,31 @@ function handleError (reply, error, cb) {
     return
   }
 
-  const result = func(error, reply.request, reply)
-  if (result !== undefined) {
-    if (result !== null && typeof result.then === 'function') {
-      wrapThenable(result, reply)
-    } else {
-      reply.send(result)
+  try {
+    const result = func(error, reply.request, reply)
+    if (result !== undefined) {
+      if (result !== null && typeof result.then === 'function') {
+        wrapThenable(result, reply)
+      } else {
+        reply.send(result)
+      }
+    }
+  } catch (err) {
+    reply[kReplyIsError] = true
+    try {
+      if (err.cause) {
+        throw new Error('Child error already has a cause set')
+      }
```

Comment:
Curious, should we throw?

What are the expectations here, if let's say, I throw a custom `Error` that already has a `cause` property already set?

Reply:
Feels unintuitive to only sometimes edit the cause property. But overwriting it and losing data sometimes seems worse.

I initially suggested we add it as a symbol property, but cause was preferred: https://github.com/fastify/fastify/pull/5222#discussion_r1433848321